llama_index/finetuning/types.py

```diff
@@ -29,3 +29,15 @@ def finetune(self) -> None:
     @abstractmethod
     def get_finetuned_model(self, **model_kwargs: Any) -> BaseEmbedding:
         """Gets finetuned model."""
+
+
+class BaseCrossEncoderFinetuningEngine(ABC):
+    """Base Cross Encoder Finetuning Engine"""
+
+    @abstractmethod
+    def finetune(self) -> None:
+        """Goes off and does stuff."""
+
+    @abstractmethod
+    def push_to_hub(self, repo_id: Any) -> None:
```

Comment:
Can't we keep the original `get_finetuned_model()` method, but just do this, and keep the push to hub method?

```
rerank = SentenceTransformerRerank(
    model="bpHigh/Cross-Encoder-LLamaIndex-Demo-v2", top_n=3
)
``` 

In fact, both the other base fine-tuning class could have a push to hub method

Reply:
The 'push_to_hub' method cannot be implemented for non open source llms and embedding models like OpenAI LLMs and cohere embeddings respectively. So it is a tough decision whether to add it as an abstract method in the Base finetuning classes or not.

As far as keeping the original 'get_finetuned_model()' wouldn't it be a bad practice as it would be declared as an abstract method and I cannot implement it in the Sentence Transformer based CrossEncoder finetuning class as it is not currently supported through the sentence transformer library? Did in the get_finetuned_model you wanted to say that I pass in the model repo to return the reranker :- rerank = SentenceTransformerRerank(
    model="bpHigh/Cross-Encoder-LLamaIndex-Demo-v2", top_n=3
) like this?  If so I agree it would be better.

Reply:
Yea exactly like that. Just to try and keep the methods on the fine tuning engines consistent 👍